primer/brand
# AnchorNav component

## Summary

Adds an accessible `AnchorNav` React component.

- Closes https://github.com/github/primer/issues/1461

Try it out:

🔗 [Storybook](https://primer-a2f43de4b8-26139705.drafts.github.io/storybook/?path=/story/components-anchornav--playground)
🔗 [Documentation](https://primer-a2f43de4b8-26139705.drafts.github.io/components/AnchorNav)


https://user-images.githubusercontent.com/13340707/205906060-132df3a6-90d1-408e-9dd3-a95d1fe2b12f.mp4




## List of notable changes:



- **added** AnchorNav pattern, along with storybook, docs and visual tests
- **added** Updated Resources Hub example with new AnchorNav pattern
- **fixed** Incorrect linter config in monorepo. Now applies IDE linting correctly in react package.
- **added** `isLarge` prop to `useWindow` hook

## What should reviewers focus on?

- Spotting issues or bugs 

## Steps to test:



1. Open [Storybook](https://primer-a2f43de4b8-26139705.drafts.github.io/storybook/?path=/story/components-anchornav--playground) examples and try it out (make sure you use iframe mode for full page experience, Intersection Observer is a little buggy in the chrome view.
1. Open [Docs](https://primer-a2f43de4b8-26139705.drafts.github.io/components/AnchorNav) page
1. Try out the examples in narrow and wide viewports

## Supporting resources (related issues, external links, etc):

- [Figma](https://www.figma.com/file/BJ95AjraesmRCWsKA013GS/Primer-Brand?node-id=2491%3A29927)

## Contributor checklist:


## Reviewer checklist:


## Screenshots:

### Keyboard support for disclosure pattern

https://user-images.githubusercontent.com/13340707/205885376-a0695734-1849-41ab-bd3d-4ab6ff1cb958.mp4

### Resources Hub example

https://user-images.githubusercontent.com/13340707/205885633-04601c3a-c92d-414d-bea3-fa82b51bbd17.mov

### Wide viewport experience
![wide desktop experience](https://user-images.githubusercontent.com/13340707/205886084-a0bcc8c4-d6b6-40c3-afe8-3ed9e146330b.png)

### Narrow viewport experience (closed)
![AnchorNav menu closed](https://user-images.githubusercontent.com/13340707/205885890-862e7365-4cf7-4fb6-87f7-73a01cbbd722.png)

### Narrow viewport experience (open)
![AnchorNav open](https://user-images.githubusercontent.com/13340707/205885846-30d0dea3-a29c-408d-9967-755b08d5217f.png)
